exchangelib/account.py

```diff
@@ -159,6 +160,41 @@ def recoverable_deleted_items(self):
     def domain(self):
         return get_domain(self.primary_smtp_address)
 
+    def export(self, ids):
+        """
+        Return export strings of the given items
+
+        Arguments:
+        'ids' is an iterable containing ItemId objects for what we want to export
+
+        Returns:
+        A list of tuples with two elements:
+            0: An ItemId, the id of the object
+            1: A string, the exported representation of the object
+        """
+        return [(id_, export) for id_, export in zip(ids, ExportItems(self).call(ids))]
+
+    def upload(self, data, folders):
+        """
+        Adds objects retrieved from export into the given folders
+
+        Arguments:
+        'data' is an iterable containing the string outputs of exports
+        'folders' is an equal lengthed iterable containing the folder that each item is to be inserted in
+
+        Returns:
+        A list of ItemIds of the new items inserted
+
+        Example:
+        account.upload(["AABBCC...", "XXYYZZ...", "ABCXYZ..."],
+                       [account.inbox, account.inbox, account.calendar])
+        -> [ItemId(...), ItemId(...), ItemId(...)]
+        """
+        merged_data = ((folder, data_str) for folder, data_str in zip(folders, data))
```

Comment:
Since the first thing you are doing is changing the input data, are there any downsides to simply expecting an iterable of tuples? For me, it seems more intuitive to keep the folder item as close to the matching data as possible in the input, and more closely matches what the service expects, and is consistent with what `bulk_update()` requires.

Another possibility would be to require a dict of `folder -> list_of_data_items`. That removes the repetition of folder items but may be less intuitive for someone who is familiar the `UploadItem`/`ExportItems` services (I'm not).